lib/daru/index.rb

```diff
@@ -330,5 +341,11 @@ def values
     def inspect
       "Daru::MultiIndex:#{self.object_id} (levels: #{levels}\nlabels: #{labels})"
     end
+
+    def factor_out
+      # Remove levels not needed for display
+      useless = levels.index { |l| l.length > 1 }
+      drop_left_level useless
+    end
```

Comment:
I can see that you've used this method to remove the left most level on [this line](https://github.com/v0dro/daru/pull/54/files#diff-8129d70abcb202c11af77e9269873527R214) and have kept it here for a uniform interface between indexes.

But according to the previous functionality, the number of levels that were removed were equal to the [number of levels](https://github.com/v0dro/daru/pull/54/files#diff-8129d70abcb202c11af77e9269873527L215) that were specified by the user.

For example:
``` ruby
d =        Daru::MultiIndex.from_tuples([
  [:c,:one,:bar],          
  [:c,:one,:baz],          
  [:c,:two,:foo],          
  [:c,:two,:bar]          
])            
#=> Daru::MultiIndex:96634260 (levels: [[:c], [:one, :two], [:bar, :baz, :foo]]
#labels: [[0, 0, 0, 0], [0, 0, 1, 1], [0, 1, 2, 0]])
v = Daru::Vector.new([1,2,3,4], index: d)
#<Daru::Vector:101434370 @name = nil @size = 4 >
#                              nil
#[:c, :one, :bar]                1
#[:c, :one, :baz]                2
#[:c, :two, :foo]                3
#[:c, :two, :bar]                4

v[:c, :one]
#<Daru::Vector:101291480 @name = nil @size = 2 >
#          nil
#[:bar]      1
#[:baz]      2

v[:c]
#=> 
#<Daru::Vector:101176290 @name = nil @size = 4 >
#                      nil
#[:one, :bar]            1
#[:one, :baz]            2
#[:two, :foo]            3
#[:two, :bar]            4
```

I don't quite understand how the correct number of levels are being dropped in spite of you nowhere specifying the number of levels the user has asked for.